core/src/main/java/io/hyperfoil/core/impl/statistics/StatisticsStore.java

```diff
@@ -84,12 +85,309 @@ public void record(String address, int phaseId, int stepId, String metric, Stati
+            jGenerator.writeNumberField("BlockedTime",summary.blockedTime);
+
+            jGenerator.writeEndObject();
+         }
+      }
+      jGenerator.writeEndArray();//end series
+   }
+   public void totalArray(JsonGenerator jGenerator, Data[] dataList,Function<Data,StatisticsSummary> getSummary, BiConsumer<JsonGenerator,Data> also) throws IOException{
+      jGenerator.writeStartArray();
+      for(Data data : dataList){
+         StatisticsSummary summary = getSummary.apply(data);//data.total.summary(percentiles);
+         if(summary==null){
+            continue;
+         }
+
+         jGenerator.writeStartObject();
+         jGenerator.writeStringField("Phase",data.phase);
+         jGenerator.writeStringField("Metric",data.metric);
+         jGenerator.writeNumberField("Start",data.total.histogram.getStartTimeStamp());
+         jGenerator.writeNumberField("End",data.total.histogram.getEndTimeStamp());
+
+         jGenerator.writeNumberField("Requests",summary.requestCount);
+         jGenerator.writeNumberField("Responses",summary.responseCount);
+         jGenerator.writeNumberField("Mean",summary.meanResponseTime);
+         jGenerator.writeNumberField("Min",summary.minResponseTime);
+         Long values[] = summary.percentileResponseTime.values().toArray(new Long[0]);
+         for(int i=0; i<percentiles.length; i++){
+            double p = percentiles[i];
+            double amt = values[i];
+            jGenerator.writeNumberField("p"+(p*100),amt);
+         }
+
+         jGenerator.writeNumberField("Max",summary.minResponseTime);
+         jGenerator.writeNumberField("MeanSendTime",summary.meanSendTime);
+         jGenerator.writeNumberField("ConnFailure",summary.connectFailureCount);
+         jGenerator.writeNumberField("Reset",summary.resetCount);
+         jGenerator.writeNumberField("Timeouts",summary.timeouts);
+         jGenerator.writeNumberField("2xx",summary.status_2xx);
+         jGenerator.writeNumberField("3xx",summary.status_3xx);
+         jGenerator.writeNumberField("4xx",summary.status_4xx);
+         jGenerator.writeNumberField("5xx",summary.status_5xx);
+         jGenerator.writeNumberField("Other",summary.status_other);
+         jGenerator.writeNumberField("Invalid",summary.invalid);
+         jGenerator.writeNumberField("CacheHits",summary.cacheHits);
+         jGenerator.writeNumberField("BlockedCount",summary.blockedCount);
+         jGenerator.writeNumberField("BlockedTime",summary.blockedTime);
+
+         if(also!=null){
+            also.accept(jGenerator,data);
+         }
+
+         jGenerator.writeEndObject();
+      }
+      jGenerator.writeEndArray();//end total.csv
+   }
+   public <T> void phaseIterFor(JsonGenerator jGenerator,
+                                T[] sorted,
+                                Function<T,String> toString,
+                                BiConsumer<String,T> consumer,
+                                Consumer<String> onNewPhase,
```

Comment:
I wonder if it could look better had you moved all the lambdas to an object with multiple methods...

Reply:
I thought about an interface with default methods and using anonymous instances. If we would rather a `PhaseIterForkWalker` then I can certainly refactor to use that instead

Reply:
Why anonymous instances? I would just have them in a class instead of a 200-line method.

Reply:
Does this look better?
https://github.com/willr3/Hyperfoil/blob/all.json/core/src/main/java/io/hyperfoil/core/impl/statistics/StatisticsStore.java#L314